QuizApp/Data/Common/Helpers/URLSession+Extension.swift

```diff
@@ -0,0 +1,47 @@
+import Foundation
+
+extension URLSession {
+
+    @available(iOS, deprecated: 15.0, message: "This extension is no longer necessary. Use API built into SDK")
+    func data(from url: URL) async throws -> (Data, URLResponse) {
+        try await withCheckedThrowingContinuation { continuation in
+            let task = dataTask(with: url) { data, response, error in
+
+                guard
+                    let data = data,
+                    let response = response
+                else {
+                    let error = error ?? NetworkError.responseCorrupted
+
+                    return continuation.resume(throwing: error)
+                }
+
+                continuation.resume(returning: (data, response))
+            }
+
+            task.resume()
+        }
+    }
+
+    @available(iOS, deprecated: 15.0, message: "This extension is no longer necessary. Use API built into SDK")
+    func data(with request: URLRequest) async throws -> (Data, URLResponse) {
+        try await withCheckedThrowingContinuation { continuation in
+            let task = self.dataTask(with: request) { data, response, error in
+
+                guard
+                    let data = data,
+                    let response = response
+                else {
+                    let error = error ?? NetworkError.responseCorrupted
+
+                    return continuation.resume(throwing: error)
+                }
+
+                continuation.resume(returning: (data, response))
+            }
+
+            task.resume()
+        }
+    }
```

Comment:
Why do you need this?

Reply:
async await for URLSession.dataTask is only available in iOS 15, this backports it